Lib/distutils/tests/test_dist.py

```diff
@@ -490,6 +490,14 @@ def test_show_help(self):
                   if line.strip() != '']
         self.assertTrue(output)
 
+    def test_parse_command_opts(self):
```

Comment:
Can you add a test case that covers:
- If a command has a global option (except help) but also defines it as
a local command option, it will not be reapplied to the distribtuion
object.

Reply:
Additionally, can you add test cases for the --help option, in which both the points in the PR description do not apply?

Reply:
@epicfaace your suggested test case surfaced a very interesting use case.

I tried to find global options also defined as command options:

```
grep -rE "verbose|quiet|dry\-run|no\-user\-cfg" Lib/distutils/command/*
```

The result of this command shows that there's only one command that includes one of the standard options as a command option, that is `bdist_rpm` with the `quiet` option.

This case is special because `quiet` is a (negative) alias for the global option `verbose`, but it's also a `bdist_rpm` command option. At this point of the command line parsing, options have already been translated to their associated Python identifiers, so the `quiet` option has been translated to the `verbose` Python identifier as a result of the global `{'quiet': 'verbose'}` alias translation. In order to match it with the command option `quiet` I would need to check for both `verbose` and its associated alias `quiet` to be present in the command options set.

Long story short: I think there's an issue, not introduced by this PR, in the way negative options aliases are interacting with the command options parsing (`bdist_rpm`'s `quiet` option case) and I can't find another case that is worth it testing.